rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -345,8 +343,31 @@ __declare_parameter_common(
   // Use the value from the overrides if available, otherwise use the default.
   const rclcpp::ParameterValue * initial_value = &default_value;
   auto overrides_it = overrides.find(name);
+
+  // If override for parameter value, use it
+  // If parameter override only, don't override descriptor
+  // If override for descriptor, use it
+  // If descriptor override only, don't override parameter value
+
   if (!ignore_override && overrides_it != overrides.end()) {
-    initial_value = &overrides_it->second;
+    auto has_parameter_override = false;
+    auto has_descriptor_override = false;
+
+    if (overrides_it->second.value.get_type() != rclcpp::ParameterType::PARAMETER_NOT_SET) {
+      has_parameter_override = true;
+    }
+    if (overrides_it->second.descriptor.name != "") {
+      has_descriptor_override = true;
+    }
+
+    if (has_parameter_override && has_descriptor_override) {
+      initial_value = &overrides_it->second.value;
+      parameter_infos.at(name).descriptor = overrides_it->second.descriptor;
+    } else if (has_parameter_override) {
+      initial_value = &overrides_it->second.value;
+    } else if (has_descriptor_override) {
+      parameter_infos.at(name).descriptor = overrides_it->second.descriptor;
+    }
```

Comment:
@bpwilcox isn't this equivalent to:
```suggestion
    if (overrides_it->second.value.get_type() != rclcpp::ParameterType::PARAMETER_NOT_SET) {
      initial_value = &overrides_it->second.value;
    }
    if (overrides_it->second.descriptor.name != "") {
      parameter_infos[name].descriptor = overrides_it->second.descriptor;
    }
```